tests/test_feedexport.py

```diff
@@ -443,7 +443,160 @@ def test_export_encoding(self):
             'csv': u'foo\r\nTest\xd6\r\n'.encode('latin-1'),
         }
 
-        for format in formats:
-            settings = {'FEED_FORMAT': format, 'FEED_EXPORT_ENCODING': 'latin-1'}
+        settings = {'FEED_EXPORT_INDENT': None, 'FEED_EXPORT_ENCODING': 'latin-1'}
+        for format, expected in formats.items():
+            settings['FEED_FORMAT'] = format
             data = yield self.exported_data(items, settings)
-            self.assertEqual(formats[format], data)
+            self.assertEqual(expected, data)
+
+    @defer.inlineCallbacks
+    def test_export_indentation(self):
+        items = [dict({'foo': ['bar']})]
+
+        output = [
+            # JSON
+            {
+                'format': 'json',
+                'indent': None,
+                'expected': b'[\n{"foo": ["bar"]}\n]',
+            },
+            {
+                'format': 'json',
+                'indent': -1,
+                'expected': b"""
+[
+{
+"foo": [
+"bar"
+]
+}
+]
+""",
+            },
+            {
+                'format': 'json',
+                'indent': 0,
+                'expected': b"""
+[
+{
+"foo": [
+"bar"
+]
+}
+]
+""",
+            },
+            {
+                'format': 'json',
+                'indent': 2,
+                'expected': b"""
+[
+{
+  "foo": [
+    "bar"
+  ]
+}
+]""",
```

Comment:
That newline is there only to make the tests more pretty, it does not end up in the final expected result (note the `strip` call in https://github.com/scrapy/scrapy/pull/2456/files#diff-0f004a6e06393cc5e29012b83956eed2R632).
Up until now (before your comment), the square brackets in the JSON export had their own line (even with `indent=0`), this PR didn't modify that. But now that you mentioned it, I amended the latest commit to make it fully compliant with the difference between `indent=None` and `indent=<integer>`. The only newline I don't know how to remove when `indent=None` is the one after the XML declaration tag, but I don't think that's a problem.